simGL/simGL.py

```diff
@@ -15,32 +16,30 @@ def extract_genotype_matrix(data):
     else:
         sys.exit("Incorrect data format")
 
-def depth_per_haplotype(rng, mean_depth, std_depth, n_ind):
-    if type(mean_depth) == int or type(mean_depth) == float:
+def depth_per_haplotype(rng, mean_depth, std_depth, n_hap, ploidy):
+    if (type(mean_depth) == int or type(mean_depth) == float) and mean_depth > 0.0:
         if type(std_depth) == int or type(std_depth) == float:
             DPh = []
-            while len(DPh) < n_ind:
-                dp = rng.normal(loc = mean_depth/2, scale = std_depth, size=1)[0]
+            while len(DPh) < n_hap:
+                dp = rng.normal(loc = mean_depth/ploidy, scale = std_depth, size=1)[0]
                 if dp > 0:
                     DPh.append(dp)
             return DPh
-    elif type(mean_depth) == np.ndarray and len(mean_depth.shape) == 1 and mean_depth.shape[0] == n_ind and (mean_depth > 0).sum() == n_ind:
+    elif type(mean_depth) == np.ndarray and len(mean_depth.shape) == 1 and mean_depth.shape[0] == n_hap and (mean_depth > 0).sum() == n_hap:
         return mean_depth
     else:
-        sys.exit("Incorrect mean_depth format")
+        raise Exception('Incorrect mean_depth format')
```

Comment:
This should probably raise a https://docs.python.org/3/library/exceptions.html#TypeError here, rather than Exception.